spring-cloud-stream/src/main/java/org/springframework/cloud/stream/aggregate/ModuleAggregationUtils.java

```diff
@@ -0,0 +1,154 @@
+import org.springframework.cloud.stream.binding.BindingUtils;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.integration.channel.DirectChannel;
+
+/**
+ * @author Marius Bogoevici
+ */
+public class ModuleAggregationUtils {
+
+	private static final Log log = LogFactory.getLog(ModuleAggregationUtils.class);
+
+	/**
+	 * Supports the aggregation of {@link Source}, {@link Sink} and {@link Processor} modules by instantiating and
+	 * binding them directly
+	 *
+	 * @param parentArgs arguments for the parent (prefixed with '--')
+	 * @param modules a list module classes to be aggregated
+	 * @param moduleArgs arguments for the modules (prefixed with '--")
+	 * @return
+	 */
+	public static ConfigurableApplicationContext runAggregated(String[] parentArgs,Class<?>[] modules, String[][] moduleArgs) {
+		ConfigurableApplicationContext parentContext = createParentContext(parentArgs != null ? parentArgs : new String[0]);
+		runEmbedded(parentContext, modules, moduleArgs);
+		return parentContext;
+	}
+
+	public static ConfigurableApplicationContext runAggregated(Class<?>... modules) {
+		return runAggregated(null, modules, null);
+	}
+
+	/**
+	 * Embeds a group of modules into an existing parent context
+	 *
+	 * @param parentContext the parent context
+	 * @param modules a list of classes, representing root context definitions for modules
+	 * @param args arguments for the modules
+	 */
+	public static void runEmbedded(ConfigurableApplicationContext parentContext, Class<?>[] modules,
+																 String[][] args) {
+		SharedChannelRegistry bean = parentContext.getBean(SharedChannelRegistry.class);
+		prepareSharedChannelRegistry(bean, modules);
+		// create child contexts first
+		LinkedHashMap<String, ConfigurableApplicationContext> children = createChildContexts(parentContext, modules, args);
+		for (Map.Entry<String, ConfigurableApplicationContext> childContextEntry : children.entrySet()) {
+			ConfigurableApplicationContext childContext = childContextEntry.getValue();
+			parentContext.getBeanFactory().registerSingleton(childContextEntry.getKey(),
+					new BindableContextWrapper(childContext));
+			childContext.setParent(parentContext);
+		}
+	}
+
+	private static ConfigurableApplicationContext createParentContext(String[] args) {
+		SpringApplicationBuilder aggregatorParentConfiguration
+				= new SpringApplicationBuilder();
+		aggregatorParentConfiguration.sources(AggregatorParentConfiguration.class)
+				.web(false)
+				.headless(true)
+				.properties("spring.jmx.default-domain=" + UUID.randomUUID().toString().replace("-", ""))
+				.listeners(new UnbindOnCloseApplicationListener());
+		return aggregatorParentConfiguration.run(args);
+	}
+
+	private static LinkedHashMap<String, ConfigurableApplicationContext> createChildContexts(ConfigurableApplicationContext applicationContext, Class<?>[] modules,
```

Comment:
For better clarity, can we make the `applicationContext` to `parentContext` here?